tools/c7n_azure/c7n_azure/policy.py

```diff
@@ -166,8 +172,16 @@ def extract_properties(options, name, properties):
             result['name'] = ResourceIdParser.get_resource_name(settings)
             result['resource_group_name'] = ResourceIdParser.get_resource_group(settings)
         else:
+            #get nested keys
             for key in properties.keys():
-                result[key] = settings.get(StringUtils.snake_to_camel(key), properties[key])
+                value = settings.get(StringUtils.snake_to_camel(key), properties[key])
+                if isinstance(value, dict):
+                    result[key] = {}
+                    for key2 in properties.get(key).keys():
+                        result[key][key2] = settings.get(StringUtils.snake_to_camel(key), properties[key]).get(
```

Comment:
Why do we need all of these keys?

Reply:
it looks something like this....
servicePlan:
            name: test
            location: East US
            skuTier: Standard
            skuName: S1
            resourceGroupName: "test"
            autoScale: 
              enableAutoScale: True
              minCapacity: 1
              maxCapacity: 1
              defaultCapacity: 1
The here gets any nested keys

Reply:
^ formatting didnt work so well.. but essentially autoscale is under service plan, and enable_autoscale, min, max, default, are under autoscale